### PR TITLE
Patch protobuf and block mode

### PR DESCRIPTION
**Pull request by RogerKSI**

### CODE REVIEW

**Review by warittornc — COMMENTED** [review]
Maybe we should add a helper function to help confirm transactions?

**Comment by RogerKSI** [comment]
> Maybe we should add a helper function to help confirm transactions?

Already reimplemented block mode to use sync mode.

**Review by taobun — APPROVED** [review]

File: pyband/client.py [hunk 1]
```diff
@@ -204,22 +207,31 @@ async def send_tx_async_mode(self, tx_bytes: bytes) -> TxResponse:
...
+        tx_hash = resp.tx_response.txhash
+        start_time = time.time()
+        while True:
+            try:
+                tx_response = await self.get_tx_response(tx_hash)
+                return tx_response
+            except Exception:
+                if time.time() - start_time >= timeout:
+                    raise
+                time.sleep(poll_interval)
```
Comment (warittornc) [inline pyband/client.py:234]:
```suggestion
                asyncio.sleep(poll_interval)
```

Comment (RogerKSI) [reply pyband/client.py]: fixed

File: pyband/client.py [hunk 2]
```diff
@@ -204,22 +207,31 @@ async def send_tx_async_mode(self, tx_bytes: bytes) -> TxResponse:
...
-        return resp.tx_response
+        if resp.tx_response.code != 0:
+            return resp.tx_response
+
+        tx_hash = resp.tx_response.txhash
+        start_time = time.time()
+        while True:
+            try:
+                tx_response = await self.get_tx_response(tx_hash)
+                return tx_response
```
[the remark this answers is not in the document]
Comment (RogerKSI) [reply pyband/client.py]: fixed

**Review by warittornc — APPROVED** [review]

**Pull request closed without merging**